controllers/config_helper.go

```diff
@@ -0,0 +1,192 @@
+package controllers
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/config/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+var errConfigCrdNotReady = errors.New("Config CRD is not ready")
+
+// Default config data
+var configDefaultResource = []byte(`
+apiVersion: config.gatekeeper.sh/v1alpha1
+kind: Config
+metadata:
+  name: config
+spec:
+  match:
+  - excludedNamespaces: ["kube-*", "multicluster-engine", "hypershift", "hive", "rhacs-operator", "open-cluster-*"] 
+    processes: ["audit", "webhook", "sync"]
+`)
+
+func createConfigFromYaml(ctx context.Context, c client.Client, namespace string) error {
+	config := &v1alpha1.Config{}
+
+	err := yaml.Unmarshal(configDefaultResource, config)
+	if err != nil {
+		return err
+	}
+
+	config.ObjectMeta = metav1.ObjectMeta{
+		Name:      "config",
+		Namespace: namespace,
+	}
+
+	err = c.Create(ctx, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkConfigHasExemptNs(ctx context.Context,
+	c client.Client, existingConfig v1alpha1.Config, log logr.Logger,
+) error {
+	configDefault := &v1alpha1.Config{}
+
+	err := yaml.Unmarshal(configDefaultResource, configDefault)
+	if err != nil {
+		return err
+	}
+
+	// Check only ExcludedNamespaces not processes
+	// Users can edit processes
+
+	for _, match := range existingConfig.Spec.Match {
+		if reflect.DeepEqual(match,
+			configDefault.Spec.Match[0]) {
+			log.Info("Default excludedNamespaces exist already")
+
+			return nil
+		}
+	}
+
+	existingConfig.Spec.Match = append(existingConfig.Spec.Match, configDefault.Spec.Match[0])
+
+	err = c.Update(ctx, &existingConfig, &client.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	log.Info("Update config because Default excludedNamespaces were missed")
+
+	return nil
+}
+
+func (r *GatekeeperReconciler) handleConfigController(mainCtx context.Context) error {
+	isCRDReady, err := checkConfigCrdAvailable(mainCtx, r.DynamicClient)
+	if err != nil {
+		return err
+	}
+
+	if !isCRDReady {
+		return errCpsCrdNotReady
+	}
+
+	if r.isConfigCtrlRunning {
+		return nil
+	}
+
+	configMgr, err := ctrl.NewManager(r.KubeConfig, ctrl.Options{
+		Scheme: r.Scheme,
+		Metrics: server.Options{
+			BindAddress: "0",
+		},
+		LeaderElection:   r.EnableLeaderElection,
+		LeaderElectionID: "config.gatekeeper.operator",
```

Comment:
This is different format than the other leader election IDs. Could you please make them have the same suffix so it's easier to identify?